.github/workflows/benchmark.yml

```diff
@@ -0,0 +1,181 @@
+name: Benchmark Xdebug performance
+
+on:
+    schedule:
+        # Every Sunday at 03:00 UTC
+        - cron: '0 3 * * 0'
+    workflow_dispatch: 
+    pull_request:
+        types: [opened, edited, reopened, synchronize]
+
+jobs:
+    run-benchmark:
+        if: |
+            github.event_name == 'schedule' || github.event_name == 'workflow_dispatch' ||
+            (
+                github.event_name == 'pull_request' &&
+                (
+                    startsWith(github.event.pull_request.title, 'perf:') ||
+                    startsWith(github.event.pull_request.title, 'Perf:') ||
+                    startsWith(github.event.pull_request.title, 'Performance:') ||
+                    startsWith(github.event.pull_request.title, 'performance:')
+                )
+            )
+        runs-on: ubuntu-latest
+        strategy:
+            fail-fast: false
+            matrix:
+                command: ["bench", "symfony", "rector"]
+                php: ["8.0", "8.1", "8.2", "8.3", "8.4", "8.5"]
+                xdebug_mode: ["no", "off", "develop", "coverage", "debug", "gcstats", "profile", "trace"]
+        steps:
+            -   uses: actions/checkout@v4
+
+            -   name: Setup PHP
+                uses: shivammathur/setup-php@v2
+                with:
+                    php-version: "${{ matrix.php }}"
+                    coverage: none
+                    tools: composer:v2.2.21
+
+            # ASLR can cause a lot of noise due to missed sse opportunities for memcpy
+            # and other operations, so we disable it during benchmarking.
+            -   name: Disable ASLR
+                run: echo 0 | sudo tee /proc/sys/kernel/randomize_va_space
+
+            -   name: Install valgrind
+                run: sudo apt-get update && sudo apt-get install -y valgrind
+                    
+            -   name: Compile
+                run: |
+                    ./.build.scripts/compile.sh
+                    sudo make install
+
+            -   name: create results and opcache folder
+                run: |
+                    mkdir -p results
+                    mkdir -p /tmp/opcache
+
+            -   name: install Symfony demo
+                if: matrix.command == 'symfony'   
+                run: |
+                    if [ "${{ matrix.php }}" \< "8.4" ]; then
+                        version="v2.0.2"
+                    else
+                        version="v2.7.0"
+                    fi
+                    git clone -q --branch "$version" --depth 1 https://github.com/symfony/demo.git ./symfony-demo
+                    cd symfony-demo
+                    composer install 
+                    find . -type f -name "*.php" -exec sed -i.bak -E 's/error_reporting\(.*\);/error_reporting(0);/g' {} +
```

Comment:
We do not want our tests to be skewed by any error, warning, notice or deprecation thrown by the code, so we set the error level to 0 on purpose. Unfortunately, Symfony overrides this setup and sets their own error reporting level. We fix this by changing all the places where Symfony is setting the error reporting level so that it continues to be set to 0